Kinvey/Kinvey/RealmCache.swift

```diff
@@ -123,7 +154,55 @@ internal class RealmCache<T: Persistable>: Cache<T> where T: NSObject {
     }
     
     func detach(_ results: RealmSwift.Results<Entity>, query: Query?) -> [T] {
-        return detach(results.map { $0 as! T }, query: query)
+        var results: [T] = results.map { $0 as! T }
+        if let predicate = query?.predicate as? NSComparisonPredicate,
+            let keyPathConstantValuePair = keyPathConstantValuePair(predicate: predicate),
+            let constantValue = keyPathConstantValuePair.constantValueExpression.constantValue,
+            constantValue is MKCircle || constantValue is MKPolygon
+        {
+            if let circle = constantValue as? MKCircle {
+                let center = CLLocation(latitude: circle.coordinate.latitude, longitude: circle.coordinate.longitude)
```

Comment:
I don't full comprehend this change. Let's discuss. On first glance, looks odd in the `RealmCache` class.